compose/service.py

```diff
@@ -780,14 +780,22 @@ def build_container_name(project, service, number, one_off=False):
 
 # Images
 
-
 def parse_repository_tag(s):
-    if ":" not in s:
-        return s, ""
-    repo, tag = s.rsplit(":", 1)
-    if "/" in tag:
-        return s, ""
-    return repo, tag
+    tag_sep = ":"
+    digest_sep = "@"
+
+    repo, tag, separator = s, "", tag_sep
+
+    if digest_sep in s:
+        separator = digest_sep
+        repo, tag = s.rsplit(digest_sep, 1)
+
+    elif tag_sep in s:
+        repo, tag = s.rsplit(tag_sep, 1)
+        if "/" in tag:
+            repo, tag = s, ""
+
+    return repo, tag, separator
```

Comment:
:+1: yes, I like this solution, good improvement.

If you update a commit with this approach and include a docstring, then I'm happy to merge. Thanks!